src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -277,34 +433,55 @@ public void hangup(String conferenceId) throws Exception {
-            logger.debug("Retrieving call status with string: " + conferenceId);
+            logger.debug("Retrieving call status with string: " + callId);
         }
         controlOperationsLock.lock();
         try {
-            if(!StringUtils.isNullOrEmpty(conferenceId)){
-                JsonNode response = doGet(String.format(CONFERENCE, conferenceId), JsonNode.class);
-                if (response == null) {
-                    return generateCallStatus(conferenceId, CallStatus.CallStatusState.Disconnected);
-                }
-                Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
-                if (conferenceIsActive != null && conferenceIsActive) {
-                    return generateCallStatus(getJsonProperty(response, "id", String.class), CallStatus.CallStatusState.Connected);
-                }
-            } else {
-                ArrayNode conferenceCalls = listConferenceCalls();
-                if(conferenceCalls != null && conferenceCalls.size() > 0){
-                    return generateCallStatus(getJsonProperty(conferenceCalls.get(0), "id", String.class), CallStatus.CallStatusState.Connected);
+            String deviceDialString = retrieveDeviceDialString();
+            Pattern pattern = Pattern.compile("(\\d+):(\\d+):.+$", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(callId);
+            if (!StringUtils.isNullOrEmpty(callId) && matcher.find()) {
+                Integer connectionId = Integer.parseInt(matcher.group(2));
+                try {
+                    JsonNode response = doGet(String.format(CONFERENCE, matcher.group(1)), JsonNode.class);
+                    if (response == null) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }
+                    Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
+                    if (conferenceIsActive != null) {
+                        // Generating callId by given conference response and target connectionId
+                        return generateCallStatus(generateCallId(response, connectionId, deviceDialString),
+                                conferenceIsActive ? CallStatus.CallStatusState.Connected : CallStatus.CallStatusState.Disconnected);
+                    }
+                } catch (CommandFailureException cfe) {
+                    if (cfe.getStatusCode() == 404) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }
+                } catch (UnknownDeviceConnection udc) {
+                    // if connection id provided was not found - logging a warning and moving forward to retrieve the list of active conference calls
+                    logger.warn(String.format("Unable to locate active connection with id %s. Using available connection instead.", connectionId));

```

Comment:
UnknownDeviceConnection is a local exception type, thrown by generateCallId() method, in case if conferenceId was found, but not the connectionId.
Essentially it is one of the valid scenarios, as it would move forward and populate the "Disconnected" status with provided callId